src/main/java/provider/wz/XMLWZFile.java

```diff
@@ -29,58 +29,64 @@
 import java.io.FileInputStream;
 import java.io.FileNotFoundException;
 import java.io.IOException;
+import java.nio.file.DirectoryStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
 
 public class XMLWZFile implements DataProvider {
-    private final File root;
+    private final Path root;
     private final WZDirectoryEntry rootForNavigation;
 
-    public XMLWZFile(File fileIn) {
+    public XMLWZFile(Path fileIn) {
         root = fileIn;
-        rootForNavigation = new WZDirectoryEntry(fileIn.getName(), 0, 0, null);
+        rootForNavigation = new WZDirectoryEntry(fileIn.getFileName().toString(), 0, 0, null);
         fillMapleDataEntitys(root, rootForNavigation);
     }
 
-    private void fillMapleDataEntitys(File lroot, WZDirectoryEntry wzdir) {
-        for (File file : lroot.listFiles()) {
-            String fileName = file.getName();
-            if (file.isDirectory() && !fileName.endsWith(".img")) {
-                WZDirectoryEntry newDir = new WZDirectoryEntry(fileName, 0, 0, wzdir);
-                wzdir.addDirectory(newDir);
-                fillMapleDataEntitys(file, newDir);
-            } else if (fileName.endsWith(".xml")) {
-                wzdir.addFile(new WZFileEntry(fileName.substring(0, fileName.length() - 4), 0, 0, wzdir));
+    private void fillMapleDataEntitys(Path lroot, WZDirectoryEntry wzdir) {
+        
+    	try (DirectoryStream<Path> stream = Files.newDirectoryStream(lroot)) {
+            for (Path path : stream) {
+            	String fileName = path.getFileName().toString();
+            	if(Files.isDirectory(path) && !fileName.endsWith(".img") ) {
+            		WZDirectoryEntry newDir = new WZDirectoryEntry(fileName, 0, 0, wzdir);
+                    wzdir.addDirectory(newDir);
+                    fillMapleDataEntitys(path, newDir);
+            	} else if (fileName.endsWith(".xml")) {
+            		wzdir.addFile(new WZFileEntry(fileName.substring(0, fileName.length() - 4), 0, 0, wzdir));
+            	}
             }
-        }
+        } catch (IOException e) {
+			// TODO Auto-generated catch block
+			e.printStackTrace();
```

Comment:
Should not print to standard out like this. _Instead, create a static logger field and log it as a warning.